### DISCUSSION TITLE
Intersection function

### DISCUSSION THREAD

**Discussion by stemill** [benbovy/spherely#59] [Q&A] [answered]
Is it possible/are there plans to implement an intersection function?

**Comment by jorisvandenbossche** [comment]
Certainly plans! 
The `s2geography` library we are wrapping (and which itself is a layer on top of `s2geometry`) already has support for this through what it calls a "boolean operation" (https://github.com/paleolimbot/s2geography/blob/1c67ddc62c30977fabca291fb68fd8cb62f3beaa/src/s2geography/build.cc#L96, where the `S2BooleanOperation` type can be union / intersection / difference / symmetric_difference), so it is a matter of wrapping this as a python function using pybind11.

Have you been trying out spherely? Or you have interest in doing so when it has more functionality?

**Reply by stemill** [reply]
Hey. Thanks for getting back to me. Definitely interested in trying out spherely but not done so yet. I'd be more than happy to test the intersection function if you manage to get it implemented.

**Comment by djhoese** [comment, marked as answer]
I know this is old, but it looks like this is implemented:

https://github.com/benbovy/spherely/pull/62

https://spherely.readthedocs.io/en/latest/_api_generated/spherely.intersection.html

Close this?

**Reply by jorisvandenbossche** [reply]
Indeed, this has been implemented in the meantime!
